clients/privacy-center/cypress/e2e/consent-banner.cy.ts

```diff
@@ -181,5 +123,228 @@ describe("Consent banner", () => {
+          expect(false).is.eql(true);
+        });
+      });
+    });
+
+    // TODO: it should be possible in the future to filter for experience on just country
+    describe("when experience is not provided, and geolocation is invalid", () => {
+      beforeEach(() => {
+        cy.fixture("consent/test_banner_options.json").then((config) => {
+          config.experience = undefined;
+          config.geolocation = {
+            country: "US",
+          };
+          config.options.isGeolocationEnabled = true;
+          cy.visitConsentDemo(config);
+        });
+      });
+
+      it.skip("fetches geolocation from API and renders the banner", () => {
+        // TODO: add when we have geolocation API call
+        expect(false).is.eql(true);
+      });
+
+      it.skip("hides the modal link", () => {
+        // TODO: add when we have link binding working
+        expect(false).is.eql(true);
+      });
+    });
+
+    describe("when experience is not provided, and geolocation is not provided, but geolocation is disabled", () => {
+      beforeEach(() => {
+        cy.fixture("consent/test_banner_options.json").then((config) => {
+          config.experience = undefined;
+          config.geolocation = undefined;
+          config.options.isGeolocationEnabled = false;
+          cy.visitConsentDemo(config);
+        });
+      });
+
+      it("does not render", () => {
+        cy.get("div#fides-consent-banner").should("not.exist");
+        cy.contains("button", "Accept Test").should("not.exist");
+      });
+
+      it.skip("hides the modal link", () => {
+        // TODO: add when we have link binding working
+        expect(false).is.eql(true);
+      });
+    });
+
+    describe("when experience contains no notices", () => {
+      beforeEach(() => {
+        cy.fixture("consent/test_banner_options.json").then((config) => {
+          config.experience.privacy_notices = [];
+          cy.visitConsentDemo(config);
+        });
+      });
+
+      it("does not render", () => {
+        cy.get("div#fides-consent-banner").should("not.exist");
+        cy.contains("button", "Accept Test").should("not.exist");
+      });
+
+      it.skip("hides the modal link", () => {
+        // TODO: add when we have link binding working
+        expect(false).is.eql(true);
+      });
+    });
+
+    describe("when experience delivery mechanism is link", () => {
+      beforeEach(() => {
+        cy.fixture("consent/test_banner_options.json").then((config) => {
+          config.experience.delivery_mechanism =
+            ExperienceDeliveryMechanism.LINK;
+          cy.visitConsentDemo(config);
+        });
+      });
+
+      it("does not render banner", () => {
+        cy.get("div#fides-consent-banner").should("not.exist");
+        cy.contains("button", "Accept Test").should("not.exist");
+      });
+
+      it.skip("shows the modal link", () => {
+        // TODO: add when we have link binding working
```

Comment:
I thought this PR does get link binding working enough to show the modal link?

Reply:
The modal link is showing by default, but I want to test that the onClick is updated correctly for the link, which I haven't yet implemented

Reply:
Just updated to hide the link by default using the following in our internal css file:
```
#fides-consent-modal-link {
  display: none;
}
```
and to show it via JS, when we have applicable notices using:
```
modalLinkEl.style.display = "inline"
```

Having it hidden by default makes more sense to me than trying to hide it every time there's a failure case on our end. It removes the uncertainty that we may leave a broken link artifact on customers' sites (because if it's shown, and we don't add an onClick to trigger the modal, the link won't do anything).

There are 2 caveats to this approach, however:

1. The CSS will probably apply correctly on customers' sites due to us targeting the ID itself, but there's a possibility the customer has overridden it on their end. We can offer guidance to the customer such that their CSS does not explicitly target our `fides-consent-modal-link` id.
2. Using `display:none` by default requires us to pick [a display prop](https://developer.mozilla.org/en-US/docs/Web/CSS/display) when choosing to display the link. I've chosen `inline` here, but I'm unsure if this should eventually be moved to a config var.